src/Functions/FunctionBase58Conversion.h

```diff
@@ -0,0 +1,232 @@
+                dst_pos += processed_offset;
+            }
+            std::memcpy(dst_pos, decoded.c_str(), ++outlen);
+
+            source += srclen + 1;
+            dst_pos += outlen;
+
+            dst_offsets[row] = dst_pos - dst;
+            src_offset_prev = src_offsets[row];
+            processed_size += outlen;
+        }
+
+        dst_data.resize(dst_pos - dst);
+    }
+};
+
+template <typename Func>
+class FunctionBase58Conversion : public IFunction
+{
+public:
+    static constexpr auto name = Func::name;
+
+    static FunctionPtr create(ContextPtr)
+    {
+        return std::make_shared<FunctionBase58Conversion>();
+    }
+
+    String getName() const override
+    {
+        return Func::name;
+    }
+
+    bool isVariadic() const override { return true; }
+
+    size_t getNumberOfArguments() const override { return 0; }
+
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return true; }
+
+    bool useDefaultImplementationForConstants() const override { return true; }
+
+    ColumnNumbers getArgumentsThatAreAlwaysConstant() const override { return {1}; }
+
+    DataTypePtr getReturnTypeImpl(const ColumnsWithTypeAndName & arguments) const override
+    {
+        if (arguments.size() != 1 && arguments.size() != 2)
+            throw Exception(
+                "Wrong number of arguments for function " + getName() + ":  1 or 2 expected.",
+                ErrorCodes::BAD_ARGUMENTS);
+
+        if (!isString(arguments[0].type))
+            throw Exception(
+                "Illegal type " + arguments[0].type->getName() + " of 1st argument of function " + getName() + ". Must be String.",
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+
+        if (arguments.size() == 2 && !isString(arguments[1].type))
+            throw Exception(
+                "Illegal type " + arguments[1].type->getName() + " of 2nd argument of function " + getName() + ". Must be String.",
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+
+        return std::make_shared<DataTypeString>();
+    }
+
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t input_rows_count) const override
+    {
+        const ColumnPtr column_string = arguments[0].column;
+        const ColumnString * input = checkAndGetColumn<ColumnString>(column_string.get());
+        if (!input)
+            throw Exception(
+                "Illegal column " + arguments[0].column->getName() + " of first argument of function " + getName(),
+                ErrorCodes::ILLEGAL_COLUMN);
+
+        std::string alphabet = "bitcoin";
+
+        if (arguments.size() == 2)
+        {
+            const auto * alphabet_column = checkAndGetColumn<ColumnConst>(arguments[1].column.get());
+
+            if (!alphabet_column)
+                throw Exception("Second argument for function " + getName() + " must be constant String", ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
```

Comment:
This error msg sounds more useful for users than the one in l. 205. Could you please use it also in l. 205?

Reply:
Why shall I? I updated the msg there, but it is for a different column